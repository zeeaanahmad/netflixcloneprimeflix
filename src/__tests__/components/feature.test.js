import React from "react";
import { render } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature/>", () => {
  it("renders the <Feature/> with populated data", () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited movies, TV shows, and more.</Feature.Title>
        <Feature.SubTitle>Watch anywhere. Cancel anytime.</Feature.SubTitle>
      </Feature>
    );

    expect(getByText("Unlimited movies, TV shows, and more.")).toBeTruthy();
    expect(getByText("Watch anywhere. Cancel anytime.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Feature/> with just the title", () => {
    const { container, queryByText, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited movies, TV shows, and more.</Feature.Title>
      </Feature>
    );

    expect(getByText("Unlimited movies, TV shows, and more.")).toBeTruthy();
    expect(queryByText("Watch anywhere. Cancel anytime.")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Feature/> with just the subtitle", () => {
    const { container, queryByText, getByText } = render(
      <Feature>
        <Feature.SubTitle>Watch anywhere. Cancel anytime.</Feature.SubTitle>
      </Feature>
    );

    expect(queryByText("Unlimited movies, TV shows, and more.")).toBeFalsy();
    expect(getByText("Watch anywhere. Cancel anytime.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
