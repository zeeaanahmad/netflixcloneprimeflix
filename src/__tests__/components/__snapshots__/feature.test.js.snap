// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature/> renders the <Feature/> with just the subtitle 1`] = `
<div
  class="feature__Container-sc-15ocszq-0 jmXdSU"
>
  <h2
    class="feature__SubTitle-sc-15ocszq-2 uayJI"
  >
    Watch anywhere. Cancel anytime.
  </h2>
</div>
`;

exports[`<Feature/> renders the <Feature/> with just the title 1`] = `
<div
  class="feature__Container-sc-15ocszq-0 jmXdSU"
>
  <h1
    class="feature__Title-sc-15ocszq-1 dxftIf"
  >
    Unlimited movies, TV shows, and more.
  </h1>
</div>
`;

exports[`<Feature/> renders the <Feature/> with populated data 1`] = `
<div
  class="feature__Container-sc-15ocszq-0 jmXdSU"
>
  <h1
    class="feature__Title-sc-15ocszq-1 dxftIf"
  >
    Unlimited movies, TV shows, and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-15ocszq-2 uayJI"
  >
    Watch anywhere. Cancel anytime.
  </h2>
</div>
`;
